mitodl/mit-open
# Shanbady/add subscribe button to pages

### What are the relevant tickets?

Closes #847

### Description (What does it do?)
This Pr adds a "subscribe" button to channel pages.

### How can this be tested?
1. checkout this branch.
2. populate channel pages. its possible not all of them are needed but I've been using the following sequence: 
```
./manage.py  backpopulate_resource_channels
./manage.py update_course_number_departments
./manage.py update_departments_schools
./manage.py update_offered_by
./manage.py update_platforms
```
3. Visit a channel page such as http://localhost:8063/c/offeror/xpro or http://localhost/c/department/history - If you are not logged in you should not see a subscribe button.
4. Once logged in, go back to a channel page and you should see a subscribe button. You should be able to subscribe and unsubscribe to each page.
5. you should see your full list of subscriptions at the api endpoint http://localhost:8063/api/v1/learning_resources_user_subscription/

### Screenshots
<img width="568" alt="Screenshot 2024-05-07 at 1 44 58 PM" src="https://github.com/mitodl/mit-open/assets/196425/2a1181dc-9dcc-4fbd-bc44-492913f7d380">
<img width="586" alt="Screenshot 2024-05-07 at 1 45 04 PM" src="https://github.com/mitodl/mit-open/assets/196425/6d1e9571-4072-45fc-a6a7-258c2009a8a1">
